packages/realm_dart/lib/src/init.dart

```diff
@@ -65,26 +58,39 @@ String get _exeDirName => p.dirname(Platform.resolvedExecutable);
 
 DynamicLibrary _openRealmLib() {
   final libName = _getLibName(realmBinaryName);
-  final root = _getNearestProjectRoot(Platform.script.path) ?? _getNearestProjectRoot(p.current);
-
-  // Try to open lib from various candidate paths
-  List<String> errMessages = [];
-  for (final open in [
-    () => _open(libName), // just ask OS..
-    () => _open(p.join(_exeDirName, libName)), // try finding it next to the executable
-    if (root != null) () => _open(p.join(root, 'binary', Platform.operatingSystem, libName)), // try finding it relative to project
-    () => _open(_getBinaryPath(libName)), // find it where it is installed by plugin
-  ]) {
+
+  DynamicLibrary? tryOpen(String path) {
     try {
-      return open();
-    } on Error catch (e) {
-      errMessages.add(e.toString());
+      return DynamicLibrary.open(path);
+    } on Error catch (_) {
+      return null;
     }
   }
-  throw RealmError(errMessages.join('\n'));
-}
 
-DynamicLibrary _open(String lib) => DynamicLibrary.open(lib);
+  Never throwError(Iterable<String> candidatePaths) {
+    final message = 'Could not open $libName at any of the following paths:\n${candidatePaths.join('\n')}'
+        '${isFlutterPlatform ? '\nDid you forget to add a dependency on realm package?' : 'Did you forget to run `dart run realm_dart install`?'}';
+    throw RealmError(message);
+  }
+
+  if (isFlutterPlatform) {
+    final path = _getPluginPath(libName);
+    return tryOpen(path) ?? throwError([path]);
+  } else {
+    final root = _getNearestProjectRoot(Platform.script.path) ?? _getNearestProjectRoot(p.current);
+    final candidatePaths = [
+      libName, // just ask OS..
+      p.join(_exeDirName, libName), // try finding it next to the executable
+      if (root != null) p.join(root, 'binary', Platform.operatingSystem, libName), // try finding it relative to project
+    ];
+    DynamicLibrary? lib;
+    for (final path in candidatePaths) {
+      lib = tryOpen(path);
+      if (lib != null) return lib;
+    }
```

Comment:
Could also extract this to a similar function as `tryOpen()` but take `paths` instead of `path` if you'd like.

Reply:
Since that would only be called from one place, it seems a little excessive to me.